design-documents/RefcatoringForStateCopling.md

```diff
@@ -0,0 +1,101 @@
+# Independent entities for order state
+
+The order state is currently saved in `sales_order` database but a state should be have and event history also it must be not cupeled to an order. 
+
+#### Common Problems
+
+This section describes common problems which affect the current flow of order entity manipulations.
+
+1. No possibility to update only order state or status without updating the whole entity.
+2. An order's life circle depends on its state which can be updated in an inconsistent way.
+3. As the state|status can be updated with the whole order and all related data it has more performance impact.
+4. Payment integrations use asynchronous operations, they can't update order state|status.
+5. Overloaded tables: *sales_order* - 206 columns (EE), *sales_invoice* - 66 (EE), *sales_creditmemo* - 73 (EE).
+6. No possibility to create own states for order.
+
+
+
+### Database Tables for Order State
+
+There will no forrend key between `sales_order_state` and `sales_order_state_link`
+
+### DB Design  for `sales_order_state`
+
+| id   | state   | created_at          |
+| ---- | ------- | ------------------- |
```

Comment:
Looks like this table doesn't make sense with such structure because the states are not unique and it more looks like history table. I would propose to create a separate table for states and statuses which will store only different types with states/statuses.